### PR TITLE
(cloudwatch) support high resolution query

### LINKED ISSUE #9246: [Bug] Cloudwatch undesired time aggregation
- What Grafana version are you using?
grafana 4.5.0

- What datasource are you using?
CloudWatch

- What OS are you running grafana on?
Centos 7

- What did you do?
I explored **RequestCount** metric over different ranges of time, always setting **Period** to _60s_

- What was the expected result?
Return **RequestCount** aggregated by _60s_

- What happened instead?
Aggregation occurs in intervals of `60` x `number of days selected via time picker`

## Range = **Today:**
today
_Query Inspector_
```
Data Source
AWS CloudWatch
Query Inspector
 Expand All  Copy to Clipboard
request:Object
method:"POST"
url:"api/datasources/proxy/6"
data:Object
region:"us-east-1"
action:"GetMetricStatistics"
parameters:Object
namespace:"AWS/ELB"
metricName:"RequestCount"
dimensions:Array[1]
statistics:Array[1]
extendedStatistics:Array[0]
startTime:1505397051
endTime:1505407851
period:60
```
**period:60**
**value: 77.99**

## Range = **2  days ago:**
2days
_Query Inspector_
```
request:Object
method:"POST"
url:"api/datasources/proxy/6"
data:Object
region:"us-east-1"
action:"GetMetricStatistics"
parameters:Object
namespace:"AWS/ELB"
metricName:"RequestCount"
dimensions:Array[1]
statistics:Array[1]
extendedStatistics:Array[0]
startTime:1505234987
endTime:1505407787
period:120
```
**period:120**
**value: 156.5**

## Range = **7  days ago:**
7days

_Query Inspector_
```
request:Object
method:"POST"
url:"api/datasources/proxy/6"
data:Object
region:"us-east-1"
action:"GetMetricStatistics"
parameters:Object
namespace:"AWS/ELB"
metricName:"RequestCount"
dimensions:Array[1]
statistics:Array[1]
extendedStatistics:Array[0]
startTime:1504802801
endTime:1505407601
period:420
``` 
**period:420**
**value: 554**

### PR DESCRIPTION
**Pull request by mtanda**
Fix https://github.com/grafana/grafana/issues/9246

@igorleao
@mauve

### CODE REVIEW

**Comment by mtanda** [comment]
I want to more good option name. "highResolution" is not good...

**Comment by mauve** [comment]
LGTM, regarding name I also cannot come up with a good name for the setting.

**Comment by mauve** [comment]
@mtanda also thanks for looking into this I tried it locally but ran out of time

File: pkg/tsdb/cloudwatch/cloudwatch.go [hunk 1]
```diff
@@ -274,6 +293,11 @@ func parseQuery(model *simplejson.Json) (*CloudWatchQuery, error) {
 		alias = "{{metric}}_{{stat}}"
 	}
 
+	highResolution, err := model.Get("highResolution").Bool()
```
Comment (bergquist) [inline pkg/tsdb/cloudwatch/cloudwatch.go]: When I test this PR I get an error. 

This should be `.MustBool(false)`

**Review by bergquist — CHANGES_REQUESTED** [review]

File: pkg/tsdb/cloudwatch/cloudwatch.go (hunk 1 above)
Comment (mtanda) [reply pkg/tsdb/cloudwatch/cloudwatch.go]: Thanks for check!
Fix it.

**Review by bergquist — APPROVED** [review]

**Comment by bergquist** [comment]
Awesome! :) Great work

**Comment by mtanda** [comment]
Thanks!!

**Comment by rehevkor5** [comment]
I can't find info in the docs about the "HighRes" checkbox. What does it actually do?

**Pull request merged**